nixos/modules/services/web-servers/caddy/default.nix

```
@@ -378,6 +379,15 @@ in
         [here](https://caddyserver.com/docs/caddyfile/concepts#environment-variables)
       '';
     };
+
+    validateConfigFile = mkOption {
+      type = types.bool;
+      default = true;
```

```suggestion
      default = lib.versionAtLeast config.system.stateVersion "25.05";
```

so only new installations will be affected?

---

I'll let the module maintainers chime in here, but IMO this isn't necessary. Breaking at build time is better than breaking at switch time (defaulting to `false` doesn't mean the configuration wasn't broken to start with).